src/Makefile.am

```diff
@@ -240,6 +240,9 @@ libcinnamon_la_CPPFLAGS = $(MUFFIN_CFLAGS) $(cinnamon_cflags)
 
 Cinnamon-0.1.gir: libcinnamon.la St-1.0.gir
 Cinnamon_0_1_gir_INCLUDES = Clutter-0 ClutterX11-0 CoglPango-0 Cogl-0 Meta-Muffin.0 Soup-2.4 CMenu-3.0
+if HAVE_NETWORKMANAGER
+Cinnamon_0_1_gir_INCLUDES += NM-1.0
+endif HAVE_NETWORKMANAGER
```

Comment:
Why are you adding this?

Reply:
When building Cinnamon with `--enable-introspection` flag it will produce `Cinnamon-0.1.gir` (which is XML with metadata description of the exported symbols) and `Cinnamon-0.1.typelib` (which is respective binary code) those are part of GObject Introspection bindings that allow using various features provided by the `libcinnamon` from any non-C compatible language that can consume GObject stuff, like for example JS. 

I know you have removed it in your #8688, but that might be a backwards incompatible change, because there might be software that is already dependent on NM being exposed through `Cinnamon-0.1.gir`. I'm not saying that is a good way to go about it, I'm just saying that we don't know what things exist out there (outside of Cinnamon itself) that might be broken after we removed this. I haven't seen any discussion about this on your PR nor anyone pointed out why this dependency existed in the first place. So I would be curious to find this out.

From my investigation, there is no any single applet or extension in Cinnamon itself that is using NM via Cinnamon, only directly. There were some references to it in https://github.com/linuxmint/cinnamon/blob/master/js/ui/extension.js#L25-L26, but according to this comment this is no longer the case after 4.2 release.

I'm comfortable taking this out, but I would like for maintainers to have a look and verify that we are comfortable with this change and all this makes sense. Let's see what they have to say once this reviewed and I will action this appropriately.

Reply:
That is a direct import of imports.gi.NMClient, not an import of imports.gi.Cinnamon using the utility functions from imports.gi.NM

-_-

It is okay to be "backwards incompatible" during the move from 4.2 to 4.4 with something that was never documented, *always* wrong, and not even proven to exist anywhere, which I still don't believe does exist anywhere because I can't think why anyone would think to do it.